module/arm-artifacts/shared-templates/storage-account.json

```diff
@@ -87,6 +95,82 @@
           "name": "storageContainers",
           "count": "[if(not(empty(parameters('storageContainerNames'))), length(parameters('storageContainerNames')), 1)]"
       }
+    },
+    {
+      "condition": "[parameters('enabledAdlsPrivateEndpoint')]",
+      "type": "Microsoft.Resources/deployments",
```

Comment:
What's the reasoning behind this being a nested deployment, rather than just another resource?

Reply:
To workaround the conflicts you normally get when trying to make a looped resource conditional, whereby the the `copyIndex` is attempted to be resolved before the condition.  I think there is another workaround where you fudged the loop count based on the condition, but this always felt cleaner (albeit more verbose!).